### PR TITLE
[14.0][MIG] sale_product_set_layout

### PR DESCRIPTION
**Pull request by rousseldenis**
- [ ] Have to do something with sequence (but maybe in sale_product_set)

### CODE REVIEW

**Review by ajaniszewska-dev — APPROVED** [review]

**Comment by rousseldenis** [comment]
@manuelregidor Could you review this ?

File: sale_product_set_layout/models/product_set.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from odoo import fields, models
+
+
+class ProductSet(models.Model):
```
Comment (manuelregidor) [inline sale_product_set_layout/models/product_set.py:4]: Shouldn't it be ProductSetLine?

File: sale_product_set_layout/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# from . import test_product_set_layout
```
Comment (manuelregidor) [inline sale_product_set_layout/tests/__init__.py:1]: Tests are not launched

File: sale_product_set_layout/views/product_set.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='set_line_ids']//field[@name='product_id']"
+                position="attributes"
+            >
+                <attribute
+                    name="attrs"
+                >{'required': [('display_type', '=', False)]}</attribute>
+            </xpath>
+            <xpath expr="//field[@name='set_line_ids']/tree" position="inside">
```
Comment (manuelregidor) [inline sale_product_set_layout/views/product_set.xml:16]: I think the lines in this tree view shoul be editable, so more products/notes/sections can be added from outside the form view.

File: sale_product_set_layout/views/product_set.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                        string="Add a section"
+                        context="{'default_display_type': 'line_section'}"
+                    />
+                    <create
+                        name="add_note_control"
+                        string="Add a note"
+                        context="{'default_display_type': 'line_note'}"
+                    />
+                </control>
+                <field name="display_type" />
```
Comment (manuelregidor) [inline sale_product_set_layout/views/product_set.xml:30]: Shouldn't this field be invisble?

**Review by manuelregidor — CHANGES_REQUESTED** [review]

**Pull request closed without merging**